lib/gollum-lib/filter/yaml.rb

```
@@ -0,0 +1,26 @@
+require 'yaml'
+# Extract YAML frontmatter from data and build metadata table. 
+
+class Gollum::Filter::YAML < Gollum::Filter
+  
+  # Regexp thanks to jekyll
+  YAML_FRONT_MATTER_REGEXP = %r!\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)!m
+
+  def extract(data)
+    data.gsub!(YAML_FRONT_MATTER_REGEXP) do
+      @markup.metadata ||= {}
+      begin
+        frontmatter = ::YAML.load(Regexp.last_match[1])
+        frontmatter = frontmatter.respond_to?(:to_h) ? frontmatter.to_h : Hash.new(frontmatter)
+        @markup.metadata.merge!(frontmatter)
+      rescue Psych::SyntaxError
```

Not a big fan of silently ignoring errors. Could we add a `misformed frontmatter` message to the frontmatter hash, so that would show up on the page?